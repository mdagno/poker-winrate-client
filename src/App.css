*{
  box-sizing: border-box;
}

html{
  background-color: #282c34;
  margin: 0;
  padding: 0;
}

.App-header {
  color: white;
  text-align: center;
  margin: 0;
  background-color: #475080;
  height: 90px;
  padding: 0;
  border-radius: 2px;
  width: 100%;
}

.App-header a {
  text-decoration: none;
  color: white;
  font-size: 18px;
  
}

.App-header h1 {
  margin: 0;
  padding: 8px;
  padding-top: 10px;
  font-family: sarina, sans-serif;
  font-style: normal;
  font-weight: 400;
}

a:hover{
  color: #00866F; 
}

.App-header i{
  position: absolute;
  top: 60px;
}

@media only screen  and (min-width : 1224px) {
  .fas{
    font-size: 2em;
    margin-top: 0.4em;
  }

  .App-header{
    height: 125px;
  }

  .App-header h1{
    font-size: 3em;
  }
}