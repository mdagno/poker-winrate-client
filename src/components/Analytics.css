.analysis select {
  width: 40%;
  margin-left: auto;
  margin-right: auto;
}

.analysis {
  position: absolute;
  margin: 10px;
  width: 80%;
  height: 400px;
  display: flex;
  flex-direction: column;
  text-align: center;
  margin-left: auto;
  margin-right: auto;
  color: white;
  background-color: white;
}