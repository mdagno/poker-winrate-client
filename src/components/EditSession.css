
.editSession{
  position: absolute;
  top: 120px;
  left: 42px;
  color: white;
  width: 80%;
  margin-left: auto;
  margin-right: auto;
}

.form-row{
  padding: 5px;
  display: flex;
  flex: 1;
  justify-content: flex-end;
  align-content: center;
}

.form-row > label {
  padding: .5em 1em .5em 0;
  flex: 1;
}
.form-row > input {
  flex: 2;
}

.form-row > textarea {
  flex: 2;
  height: 100px;
}