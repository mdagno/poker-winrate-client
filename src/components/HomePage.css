.homePage{
  color: #222;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 80%;
  margin-top: 50px;
  margin-left: auto;
  margin-right: auto;
  border-radius: 3px;
  border-top: 10px outset white;
  border-left: 10px outset white;
  background-color: #ccc;
  text-align: left;
}

.homePage h2 {
  text-align: left;
  margin: 0;
  padding: 0.5em;
  padding-bottom: 0.2em;
  border-bottom: 2px dashed white;
}

.homePage h3{
  margin: 0;
  padding: 1em;
  border-bottom: 1px solid white;
}

@media only screen and (min-device-width : 768px) and (max-device-width : 1024px){
  .homePage{
    text-align: center;
  }
  
}

/* Desktops and laptops ----------- */
@media only screen  and (min-width : 1224px) {
  .homePage{
    text-align: center;
    width: 40%;
    height: 50%;
    border-top: 30px outset white;
    border-left: 30px outset white;
  }
  
  .homePage h2{
    padding: 1em;
    margin-left: 1em;
  }
}