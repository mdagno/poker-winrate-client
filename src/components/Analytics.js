import React from 'react'
import './Analytics.css'
import { VictoryAxis } from 'victory';
import { VictoryLine } from 'victory';
import { VictoryTheme } from 'victory';
import { VictoryChart } from 'victory';
import ApiContext from '../ApiContext';

export default class Analytics extends React.Component {
state = {
  filter: 'Profits',
}
static contextType = ApiContext;

  generateData = () => {
    const sessionsList = this.context.sessionsList;
    let filter = this.state.filter;
    let renderedData;
    if (filter === 'Profits'){
      renderedData = sessionsList.map((session, index) => {
        return {x: index + 1, y: session.cashed_out - session.buy_in}
      })
    }
    else if(filter === 'Online'){
      let filteredData = sessionsList.filter(session => session.game_type_one === 'Online')
      renderedData = filteredData.map((session, index) => {
        return {x: index + 1, y: session.cashed_out - session.buy_in}
      })
    }
    else if(filter === 'Live'){
      let filteredData = sessionsList.filter(session => session.game_type_one === 'Live')
      renderedData = filteredData.map((session, index) => {
        return {x: index + 1, y: session.cashed_out - session.buy_in}
      })
    }
    console.log()
    
    return renderedData;
  }

  calculateDomain = () => {
    let sessionProfits = this.generateData().map(session => Math.abs(session.y))
    let yDomain = Math.max(...sessionProfits)
    return yDomain;
  }

  handleFilterChange = filterValue => {
    this.setState({
      filter: filterValue,
    })
  }

  render(){
    console.log(this.generateData())
    console.log(this.calculateDomain())
    console.log(this.state)
    return(
      <div className="analysis">
        <label for="filterAnalytics">See Data: </label>
        <select className="filterAnalytics" onChange={e => this.handleFilterChange(e.target.value)}>
          <option value="Profits">Total Profits</option>
          <option value="Online">Online Profits</option>
          <option value="Live">Live Profits</option>
        </select>
        <VictoryChart  height={500} width={450} padding={70} animate={1000} tickValues={[]}>
          <VictoryLine 
            interpolation='linear'
            data={this.generateData()} 
            style={
              {
                data: {
                  stroke: '#09d3ac', 
                  strokeWidth: 1
                },
                parent: {border: "1px solid #ccc"}
            }}
          />
        </VictoryChart>
      </div>
    )
  }
}