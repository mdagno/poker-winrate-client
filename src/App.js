import React from 'react';
import './App.css';
import SessionForm from './components/SessionForm';
import NavBar from './components/NavBar';
import SessionList from './components/SessionList';
import Session from './components/Session'
import Graph from './components/Graph'
import HomePage from './components/HomePage'
import { Route } from 'react-router-dom';
import { Link } from 'react-router-dom';
import { Switch } from 'react-router-dom';
import ApiService from './services/api-service';
import ApiContext from './ApiContext';
import config from './config'


export default class App extends React.Component {
  static contextType = ApiContext;

  componentDidMount() {
    ApiService.getSessions()
    .then(this.context.setSessionsList)
    .catch(this.context.setError)
  }


  render(){
  return (
    <div className="App">
      <header className="App-header">
        <Link to='/'>
          <h1>Poker WinRate</h1>
        </Link>
        <h2>Find your edge</h2>
      </header>

      <Switch>
      <NavBar />
      </Switch>
      <main>
        <Switch>
        <Route 
        exact path='/'
        render={(props) => <HomePage {...props}/>}
        />
        <Route 
        expact 
        path='/addsession'
        render={(props) => <SessionForm {...props}/>}
        />
        <Route 
        exact path='/sessions'
        render={(props) => <SessionList {...props} />}
        />
        <Route 
        exact path='/analysis'
        render={(props) => <Graph {...props} />}
        />
        <Route 
        exact path='/sessions/:sessionid'
        render={(props) => <Session {...props} />}
        />
        </Switch>
      </main>
    </div>
  );
  }
}
