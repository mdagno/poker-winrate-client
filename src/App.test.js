import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import SessionForm from './components/SessionForm/SessionForm';
import NavBar from './components/NavBar/NavBar';
import SessionList from './components/SessionList/SessionList';
import Session from './components/Session/Session'
import Analytics from './components/Analytics/Analytics'
import HomePage from './components/HomePage/HomePage'
import EditSession from './components/EditSession/EditSession'
import { BrowserRouter } from 'react-router-dom';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <BrowserRouter>
  <App />
  </BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe.only('The UI of the components render as expected', () => {
  it('HomePage renders correctly', () => {
    const tree = renderer
      .create(<HomePage />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('SessionList renders correctly', () => {
    const tree = renderer
      .create(<SessionList />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('SessionForm renders correctly', () => {
    const tree = renderer
      .create(<SessionForm />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('Session renders correctly', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
      <Session />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('EditSession renders correctly', () => {
    const props = { match: {params: {session_id: 1} } };
    const tree = renderer
      .create(<EditSession {...props}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
  
  it('NavBar renders correctly', () => {
    const tree = renderer
      .create(<NavBar />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('Analytics renders correctly', () => {
    const tree = renderer
      .create(<Analytics />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})


