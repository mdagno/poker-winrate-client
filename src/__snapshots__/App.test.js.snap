// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The UI of the components render as expected Analytics renders correctly 1`] = `
<div
  className="analysis"
>
  <select
    className="filterAnalytics"
    onChange={[Function]}
  >
    <option
      value="Profits"
    >
      Total Profits
    </option>
    <option
      value="Online"
    >
      Online Profits
    </option>
    <option
      value="Live"
    >
      Live Profits
    </option>
    <option
      value="Big Blind"
    >
      BB/hour
    </option>
  </select>
  <div
    className="VictoryContainer"
    style={
      Object {
        "height": "100%",
        "pointerEvents": "none",
        "position": "relative",
        "touchAction": "none",
        "userSelect": "none",
        "width": "100%",
      }
    }
  >
    <svg
      height={500}
      role="img"
      style={
        Object {
          "height": "auto",
          "pointerEvents": "all",
          "width": "100%",
        }
      }
      viewBox="0 0 450 500"
      width={450}
    >
      <g
        clipPath="url(#victory-clip-2)"
        style={
          Object {
            "border": "1px solid #ccc",
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <defs>
          <clipPath
            id="victory-clip-2"
          >
            <rect
              height={360}
              vectorEffect="non-scaling-stroke"
              width={0}
              x={70}
              y={70}
            />
          </clipPath>
        </defs>
        <path
          d={null}
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "fill": "transparent",
              "opacity": 1,
              "stroke": "#009BB0",
              "strokeWidth": 3,
            }
          }
        />
      </g>
      <text
        direction="inherit"
        dx={0}
        dy={4.97}
        x={200}
        y={470}
      >
        <tspan
          dx={0}
          style={
            Object {
              "fill": "#252525",
              "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
              "fontSize": 14,
              "parent": Object {
                "height": "100%",
                "userSelect": "none",
                "width": "100%",
              },
              "stroke": "transparent",
            }
          }
          textAnchor="start"
          x={200}
        >
          Sessions
        </tspan>
      </text>
      <text
        direction="inherit"
        dx={0}
        dy={4.97}
        x={25}
        y={30}
      >
        <tspan
          dx={0}
          style={
            Object {
              "fill": "#252525",
              "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
              "fontSize": 14,
              "parent": Object {
                "height": "100%",
                "userSelect": "none",
                "width": "100%",
              },
              "stroke": "transparent",
            }
          }
          textAnchor="start"
          x={25}
        >
          Profit
        </tspan>
      </text>
      <g
        role="presentation"
      >
        <line
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "fill": "transparent",
              "stroke": "#90A4AE",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeWidth": 2,
            }
          }
          vectorEffect="non-scaling-stroke"
          x1={70}
          x2={380}
          y1={430}
          y2={430}
        />
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={132}
            x2={132}
            y1={430}
            y2={70}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={132}
            x2={132}
            y1={430}
            y2={435}
          />
          <text
            direction="inherit"
            dx={0}
            dy={10.26}
            id="chart-axis-3-tickLabels-0"
            x={132}
            y={443}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={132}
            >
              0.2
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={194}
            x2={194}
            y1={430}
            y2={70}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={194}
            x2={194}
            y1={430}
            y2={435}
          />
          <text
            direction="inherit"
            dx={0}
            dy={10.26}
            id="chart-axis-3-tickLabels-1"
            x={194}
            y={443}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={194}
            >
              0.4
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={256}
            x2={256}
            y1={430}
            y2={70}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={256}
            x2={256}
            y1={430}
            y2={435}
          />
          <text
            direction="inherit"
            dx={0}
            dy={10.26}
            id="chart-axis-3-tickLabels-2"
            x={256}
            y={443}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={256}
            >
              0.6
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={318}
            x2={318}
            y1={430}
            y2={70}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={318}
            x2={318}
            y1={430}
            y2={435}
          />
          <text
            direction="inherit"
            dx={0}
            dy={10.26}
            id="chart-axis-3-tickLabels-3"
            x={318}
            y={443}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={318}
            >
              0.8
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={380}
            x2={380}
            y1={430}
            y2={70}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={380}
            x2={380}
            y1={430}
            y2={435}
          />
          <text
            direction="inherit"
            dx={0}
            dy={10.26}
            id="chart-axis-3-tickLabels-4"
            x={380}
            y={443}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={380}
            >
              1.0
            </tspan>
          </text>
        </g>
      </g>
      <g
        role="presentation"
      >
        <line
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "fill": "transparent",
              "stroke": "#90A4AE",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeWidth": 2,
            }
          }
          vectorEffect="non-scaling-stroke"
          x1={70}
          x2={70}
          y1={70}
          y2={430}
        />
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={380}
            y1={358}
            y2={358}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={65}
            y1={358}
            y2={358}
          />
          <text
            direction="inherit"
            dx={0}
            dy={4.26}
            id="chart-axis-4-tickLabels-0"
            x={57}
            y={358}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="end"
              x={57}
            >
              0.2
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={380}
            y1={286}
            y2={286}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={65}
            y1={286}
            y2={286}
          />
          <text
            direction="inherit"
            dx={0}
            dy={4.26}
            id="chart-axis-4-tickLabels-1"
            x={57}
            y={286}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="end"
              x={57}
            >
              0.4
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={380}
            y1={214}
            y2={214}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={65}
            y1={214}
            y2={214}
          />
          <text
            direction="inherit"
            dx={0}
            dy={4.26}
            id="chart-axis-4-tickLabels-2"
            x={57}
            y={214}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="end"
              x={57}
            >
              0.6
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={380}
            y1={142}
            y2={142}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={65}
            y1={142}
            y2={142}
          />
          <text
            direction="inherit"
            dx={0}
            dy={4.26}
            id="chart-axis-4-tickLabels-3"
            x={57}
            y={142}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="end"
              x={57}
            >
              0.8
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={380}
            y1={70}
            y2={70}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={65}
            y1={70}
            y2={70}
          />
          <text
            direction="inherit"
            dx={0}
            dy={4.26}
            id="chart-axis-4-tickLabels-4"
            x={57}
            y={70}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="end"
              x={57}
            >
              1.0
            </tspan>
          </text>
        </g>
      </g>
    </svg>
    <div
      style={
        Object {
          "height": "auto",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
          "zIndex": 99,
        }
      }
    >
      <svg
        height={500}
        style={
          Object {
            "height": "auto",
            "overflow": "visible",
            "width": "100%",
          }
        }
        viewBox="0 0 450 500"
        width={450}
      />
    </div>
  </div>
</div>
`;

exports[`The UI of the components render as expected EditSession renders correctly 1`] = `
<div
  className="editSession"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-row"
    >
      <label
        for="gameTypeOne"
      >
        Live or Online: 
      </label>
      <select
        defaultValue="Live"
        id="gameTypeOne"
        onChange={[Function]}
      >
        <option
          value="Live"
        >
          Live game
        </option>
        <option
          value="Online"
        >
          Online game
        </option>
      </select>
    </div>
    <div
      className="form-row"
    >
      <label
        for="smallBlind"
      >
        Small Blind: 
      </label>
      <input
        className="form-smallBlind"
        defaultValue={null}
        id="smallBlind"
        onChange={[Function]}
        type="number"
      />
    </div>
    <div
      className="form-row"
    >
      <label
        for="bigBlind"
      >
        Big Blind: 
      </label>
      <input
        className="form-bigBlind"
        defaultValue={null}
        id="bigBlind"
        onChange={[Function]}
        type="number"
      />
    </div>
    <div
      className="form-row"
    >
      <label
        for="buyIn"
      >
        Buy-in: 
      </label>
      <input
        className="form-buyin"
        defaultValue={null}
        id="buyIn"
        onChange={[Function]}
        type="number"
      />
    </div>
    <div
      className="form-row"
    >
      <label
        for="cashedOut"
      >
        Cashed out: 
      </label>
      <input
        className="form-cashedOut"
        defaultValue={null}
        id="cashedOut"
        onChange={[Function]}
        type="number"
      />
    </div>
    <div
      className="form-row"
    >
      <label
        for="cashedOut"
      >
        Time played: 
      </label>
      <input
        className="form-SessionLength"
        defaultValue={null}
        id="sessionLength"
        onChange={[Function]}
        type="number"
      />
    </div>
    <div
      className="form-row"
    >
      <label
        for="notes"
      >
        Notes: 
      </label>
      <textarea
        className="form-notes"
        defaultValue=""
        id="notes"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-row"
    >
      <button
        type="submit"
      >
        Save
      </button>
    </div>
  </form>
</div>
`;

exports[`The UI of the components render as expected HomePage renders correctly 1`] = `
<div
  className="homePage"
>
  <h2>
    Your Summary
  </h2>
  <h3>
    $
    0
     Total Profit
  </h3>
  <h3>
    Most Played Stake: $5/$10
  </h3>
  <h3>
    0
     Sessions Played
  </h3>
  <h3>
    NaN
     BB/hr
  </h3>
</div>
`;

exports[`The UI of the components render as expected NavBar renders correctly 1`] = `
<nav
  className="navBar"
>
  <div
    className="hamburger-menu"
  >
    <i
      class="fas fa-bars"
      onClick={[Function]}
    />
  </div>
</nav>
`;

exports[`The UI of the components render as expected Session renders correctly 1`] = `
<div
  className="detailedView"
>
  <div
    className="sessionDetails"
  >
    <h3>
      October 23rd 2019, 5:28:15 pm
    </h3>
    <ul
      className="gameTypes"
    >
      <li />
      <li />
    </ul>
    <p>
      $
       / $
    </p>
    <p>
      Time played: 
       hours
    </p>
    <ul
      className="cashMoney"
    >
      <li>
        Buy in: $
      </li>
      <li
        className="losing"
      >
        -$
        NaN
      </li>
      <li>
        Cashed out: $
      </li>
    </ul>
    <p
      id="notes"
    />
    <a
      href="/sessions/undefined/edit"
      onClick={[Function]}
    >
      <button>
        Edit
      </button>
    </a>
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;

exports[`The UI of the components render as expected SessionForm renders correctly 1`] = `
<div
  className="addSession"
>
  <form
    onSubmit={[Function]}
  >
    <legend>
      <h2>
        Enter Session Details
      </h2>
    </legend>
    <div
      className="form-row gameTypeTwo"
    >
      <label
        for="gameTypeTwo"
      >
        Game Type: 
      </label>
      <select
        defaultValue="Cash"
        id="gameTypeTwo"
        onChange={[Function]}
      >
        <option
          value="Cash"
        >
          Cash Game
        </option>
        <option
          value="Tournament"
        >
          Tournament
        </option>
      </select>
    </div>
    <fieldset>
      <div
        className="form-row"
      >
        <label
          for="gameTypeOne"
        >
          Live or Online: 
        </label>
        <select
          defaultValue="Live"
          id="gameTypeOne"
          onChange={[Function]}
        >
          <option
            value="Live"
          >
            Live game
          </option>
          <option
            value="Online"
          >
            Online game
          </option>
        </select>
      </div>
      <div
        className="form-row"
      >
        <label
          for="smallBlind"
        >
          Small Blind: 
        </label>
        <input
          className="form-smallBlind"
          id="smallBlind"
          onChange={[Function]}
          type="number"
        />
      </div>
      <div
        className="form-row"
      >
        <label
          for="bigBlind"
        >
          Big Blind: 
        </label>
        <input
          className="form-bigBlind"
          id="bigBlind"
          onChange={[Function]}
          type="number"
        />
      </div>
      <div
        className="form-row"
      >
        <label
          for="buyIn"
        >
          Buy-in: 
        </label>
        <input
          className="form-buyin"
          id="buyIn"
          onChange={[Function]}
          type="number"
        />
      </div>
      <div
        className="form-row"
      >
        <label
          for="cashedOut"
        >
          Cashed out: 
        </label>
        <input
          className="form-cashedOut"
          id="cashedOut"
          onChange={[Function]}
          type="number"
        />
      </div>
      <div
        className="form-row"
      >
        <label
          for="cashedOut"
        >
          Time played: 
        </label>
        <input
          className="form-SessionLength"
          id="sessionLength"
          onChange={[Function]}
          type="number"
        />
      </div>
      <div
        className="form-row"
      >
        <label
          for="notes"
        >
          Notes: 
        </label>
        <textarea
          className="form-notes"
          id="notes"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-row"
      >
        <button
          type="submit"
        >
          Submit
        </button>
        <button
          type="reset"
        >
          Reset
        </button>
      </div>
    </fieldset>
  </form>
</div>
`;

exports[`The UI of the components render as expected SessionList renders correctly 1`] = `
<div
  className="sessionList"
>
  <h2>
    Your Sessions
  </h2>
  <label
    for="filterSessions"
  >
    Filter By: 
  </label>
  <select
    onChange={[Function]}
  >
    <option
      value="Date"
    >
      Most Recent
    </option>
    <option
      value="Live"
    >
      Live
    </option>
    <option
      value="Online"
    >
      Online
    </option>
    <option
      value="Stake"
    >
      Stake
    </option>
    <option
      value="Time played"
    >
      Time played
    </option>
  </select>
  <ul
    className="sessionList-list"
  />
</div>
`;
